modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceModificationApi.java

```diff
@@ -0,0 +1,182 @@
+        boolean currentUserIsOwner =
+                this.userServiceHelper.currentUserIsOwner(deployServiceEntity.getUserId());
+        if (!currentUserIsOwner) {
+            throw new AccessDeniedException(
+                    "No permissions to modify services belonging to other users.");
+        }
+        if (Objects.nonNull(deployServiceEntity.getLockConfig())
+                && deployServiceEntity.getLockConfig().isModifyLocked()) {
+            String errorMsg = "Service with id " + serviceId + " is locked from modification.";
+            throw new ServiceLockedException(errorMsg);
+        }
+
+        DeployTask modifyTask =
+                this.deployService.getModifyTask(modifyRequest, deployServiceEntity);
+        UUID modifyTaskId = UUID.randomUUID();
+        deployService.modifyService(modifyTaskId, modifyTask, deployServiceEntity);
+        return modifyTaskId;
+    }
+
+    /**
+     * List service modification audits by service id and task status.
+     *
+     * @param serviceId  id of service.
+     * @param taskStatus status of the management task.
+     * @return service modification audits.
+     */
+    @Tag(name = "ServiceModification",
+            description = "APIs to manage modifications of the service instances.")
+    @Operation(description = "List modification audits of the service instance.")
+    @GetMapping(value = "/services/{serviceId}/modifications", produces =
+            MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    @AuditApiRequest(methodName = "getCspFromServiceId")
+    public List<ServiceModificationAuditDetails> listServiceModificationAudits(
+            @Parameter(name = "serviceId", description = "id of the service")
+            @PathVariable(name = "serviceId") String serviceId,
+            @Parameter(name = "taskStatus", description = "status of the modification")
+            @RequestParam(name = "taskStatus", required = false) TaskStatus taskStatus) {
+        return modificationAuditManager.listServiceModificationAudits(
+                UUID.fromString(serviceId), taskStatus);
+    }
+
+    /**
+     * Delete all state management modifications of the service.
+     *
+     * @param serviceId id of the service.
+     */
+    @Tag(name = "ServiceModification",
+            description = "APIs to manage modifications of the service instances")
+    @Operation(description = "Delete all state management modifications of the service.")
+    @DeleteMapping(value = "/services/{serviceId}/modifications",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @AuditApiRequest(methodName = "getCspFromServiceId")
+    public void deleteAuditsByServiceId(
+            @Parameter(name = "serviceId", description = "id of the service")
+            @PathVariable(name = "serviceId") String serviceId) {
+        modificationAuditManager.deleteAuditsByServiceId(UUID.fromString(serviceId));
```

Comment:
we need the userId check here as well.
For Admin role we can allow without the user id check. Applies to all other methods as well.

Reply:
Added.  I will create a new issue to add permissions check in the APIs for service state management task